pkg/kmm/containers_test.go

```diff
@@ -0,0 +1,230 @@
+package kmm
+
+import (
+	"testing"
+
+	v1 "github.com/NVIDIA/gpu-operator/api/v1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestModuleLoaderContainerBuilder(t *testing.T) {
+	testCases := []struct {
+		name          string
+		expectedError string
+	}{
+		{
+			name:          "kmod",
+			expectedError: "",
+		},
+		{
+			name:          "",
+			expectedError: "'modName' cannot be empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testModuleLoaderContainerBuilder := NewModLoaderContainerBuilder(testCase.name)
+		assert.Equal(t, testCase.expectedError, testModuleLoaderContainerBuilder.errorMsg)
+		assert.NotNil(t, testModuleLoaderContainerBuilder.definition)
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testCase.name, testModuleLoaderContainerBuilder.definition.Modprobe.ModuleName)
+		}
+	}
+}
+
+func TestModuleLoaderContainerBuilder_WithModprobeSpec(t *testing.T) {
+	testCases := []struct {
+		dirName            string
+		fwPath             string
+		parameters         []string
+		args               []string
+		rawargs            []string
+		moduleLoadingOrder []string
+	}{
+		{
+			dirName:            "",
```

Comment:
Why we allow empty values? Do we need to improve verification in WithModprobeSpec

Reply:
API allows and it fits my needs. 

If we want to move that path, I think this function should be split in 5 smaller functions that each check. Still that is outside of the scope currently